File: phoenixfront/src/app/models/archive/ArchivedStockDTO.ts
import { Campaigndto } from "../agents/Campaigndto";
import { ArchivedProductsDTO } from "./ArchivedProductsDTO";
import { ArchivedSoldProductsDTO } from "./ArchivedSoldProductsDTO";

export interface ArchivedStockDTO {
    stockReference?: string;
    productTypes?: string[];
    campaignRef?: string;
    campaigndto?: Campaigndto;
    shippingDate?: Date;
    receivedDate?: Date;
    dueDate?: Date;
    checked?: boolean;
    notes?: string;
    stockValue?: number;
    products?: ArchivedProductsDTO[];
    soldproducts?: ArchivedSoldProductsDTO[];
}

File: phoenix/stock/src/main/java/com/phoenix/archivedmodeldto/ArchivedStockDTO.java
package com.phoenix.archivedmodeldto;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ArchivedStockDTO {
    private String stockReference;
    private List<String> productTypes;
    private String campaignRef;
    private LocalDate shippingDate;
    private LocalDate receivedDate;
    private LocalDate dueDate;
    private boolean checked;
    private String notes;
    private BigDecimal stockValue;
}
